syntax/dirvish.vim

```diff
@@ -4,18 +4,14 @@ endif
 
 let s:sep = exists('+shellslash') && !&shellslash ? '\' : '/'
 let s:escape = 'substitute(escape(v:val, ".$~"), "*", ".*", "g")'
+let s:columns = join(map(dirvish#get_columns(), 'escape(v:val.mark, "*.^$~\\")'), '\|')
 
 " Define once (per buffer).
 if !exists('b:current_syntax')
-  exe 'syntax match DirvishPathHead =.*\'.s:sep.'\ze[^\'.s:sep.']\+\'.s:sep.'\?$= conceal'
+  exe 'syntax match DirvishPathHead =^\('.s:columns.'\)\?\zs.*\'.s:sep.'\ze[^\'.s:sep.']\+\'.s:sep.'\?$= conceal'
   exe 'syntax match DirvishPathTail =[^\'.s:sep.']\+\'.s:sep.'$='
   exe 'syntax match DirvishSuffix   =[^\'.s:sep.']*\%('.join(map(split(&suffixes, ','), s:escape), '\|') . '\)$='
+  exe 'syntax match DirvishArg =^\*\s.*$= contains=DirvishPathHead'
 endif
 
-" Define (again). Other windows (different arglists) need the old definitions.
-" Do these last, else they may be overridden (see :h syn-priority).
-for s:p in argv()
-  exe 'syntax match DirvishArg ,'.escape(fnamemodify(s:p,':p'),',*.^$~\').'$, contains=DirvishPathHead'
-endfor
```

Comment:
this will regress #123 ?

Reply:
Not really, it will be highlighted as before, and marked.